src/Sarif/Autogenerated/Resources.cs

```diff
@@ -41,7 +43,8 @@ public SarifNodeKind SarifNodeKind
         /// A dictionary, each of whose keys is a string and each of whose values is a 'rule' object, that describe all rules associated with an analysis tool or a specific run of an analysis tool.
         /// </summary>
         [DataMember(Name = "rules", IsRequired = false, EmitDefaultValue = false)]
-        public IDictionary<string, Rule> Rules { get; set; }
+        [JsonConverter(typeof(Microsoft.CodeAnalysis.Sarif.Readers.RuleDictionaryConverter))]
+        public IDictionary<string, IRule> Rules { get; set; }
```

Comment:
>IRule [](start = 35, length = 5)

We now emit IRule into the OM. We should either prefer this or eliminate IRule altogether. Probably should prefer the latter. this would be a large breaking change, so should just do it if we are. I think it may be preferable, otherwise we need to figure out how to code gen things like IRuleEquatable.

Reply:
@michaelcfanning I totally agree. Please remind me what the original motivation was for `IRule` -- I know you just told me recently.

---
In reply to: [221101257](https://github.com/Microsoft/sarif-sdk/pull/1103#discussion_r221101257) [](ancestors = 221101257)